src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -222,10 +222,27 @@ private bool TryReplaceBannedFunction(FunctionExpression expression, [NotNullWhe
                     throw new ArgumentException($"Expected 0 properties for binary function {expression.Function}");
                 }
 
+                var leftParameter = expression.Parameters[0];
+                var rightParameter = expression.Parameters[1];
+                // if token is = or != check to see if they are insensitive conditions i.e =~ or !~
+                if (binaryTokenType is TokenType.Equals || binaryTokenType is TokenType.NotEquals)
+                {
+                    if(leftParameter is FunctionExpression leftFunctionExpression &&
+                    rightParameter is FunctionExpression rightFunctionExpression &&
+                    leftFunctionExpression.Function == "toLower" &&
+                    rightFunctionExpression.Function == "toLower")
```

Comment:
Technically, there may be some using "toUpper" for case-insensitive comparison, although I don't have the data to know how common that is! I think it might be worth just supporting both.